apiserver/facades/controller/singular/singular.go

```diff
@@ -22,19 +23,28 @@ func NewExternalFacade(st *state.State, _ facade.Resources, auth facade.Authoriz
 		return nil, err
 	}
 
-	backend := getBackend(st, m)
+	backend := getBackend(st, m.ModelTag())
 	return NewFacade(backend, auth)
 }
 
-var getBackend = func(st *state.State, m *state.Model) Backend {
-	return struct {
-		*state.State
-		*state.Model
-	}{st, m}
+var getBackend = func(st *state.State, modelTag names.ModelTag) Backend {
+	return &stateBackend{st, modelTag}
+}
+
+type stateBackend struct {
+	*state.State
+	modelTag names.ModelTag
+}
+
+func (b *stateBackend) ModelTag() names.ModelTag {
```

Comment:
// ModelTag tells the Facade what models it should consider requests for.

Reply:
// ModelTag is part of the Backend interface